pkg/controllers/user/pipeline/controller/pipeline/pipeline.go

```diff
@@ -32,7 +32,7 @@ func Register(ctx context.Context, cluster *config.UserContext) {
 }
 
 func (l *Lifecycle) Create(obj *v3.Pipeline) (*v3.Pipeline, error) {
-	return l.sync(obj)
+	return obj, nil
```

Comment:
@gitlawr could you explain how calling out the sync logic from the create method would solve the problem? The sync call is still being made from the Updated(), and if we are trying to guard from the duplicates being created, we might want to address it in the sync method itself.

Reply:
The Create() is likely to be called twice with old revision resource while Updated() is not on creation. 
I don't find an assuring way for this interacting with a third party system in the controller manner.
1. if we do `sync{createwebhook();updatecondition(obj)}`, when update retries duplicates happens
2. if we do `sync{checkwebhook();createwebhook();updatecondition(obj)}`, it does not help when retry go faster then the creation from remote system.
3. we may try `sync{updatecondition(obj);createwebhook_on_update_done(obj);}`, I don't do in this way because updating object is handled by the lifecycle.

Reply:
@orangedeng That does not solve the cause as well. The decision for hook creation is based on pipeline resource status in lifecycle function. Even if we lock and check remote status, we still regenerate the secret for the hook in the pipeline, making them mismatch.

**Why skipping Create function works:**
In the lifecycle we do some initialization work before & after lifecycle.Create(). e.g., [set finalizer](https://github.com/rancher/norman/blob/master/lifecycle/object.go#L157), [set initialized](https://github.com/rancher/norman/blob/master/lifecycle/object.go#L169), they causes additional syncs which are not under lifecycle functions' control.(so that multiple syncs run concurrently with the pipeline status showing webhook is not registered)
When the pipeline dedicated logic goes merely to lifecycle.Updated, the resource is initialized and we are not expecting these framework syncs occur(Otherwise `cont ==false` and it does not go down to [Updated](https://github.com/rancher/norman/blob/master/lifecycle/object.go#L58)). Therefore, we can infer from pipeline condition whether its webhook is registered or not.